### DISCUSSION TITLE
Example GET requests

### DISCUSSION THREAD

**Discussion by Ericnunez** [Netflix/dgs-framework#649] [Q&A] [answered]
I want to create some sample curl requests for an api that I created using the DGS framework but I am having some trouble with the GET request. Am I able to make GET requests with url parameters? The response that I get when making a GET request is Invalid GraphQL request - no request body was provided. I imagine that with url parameters this won't make sense but I have seen other graphql APIs offer GET support.

Sample curl request sans encoding:
 curl -g \
 -X GET \
 -H "Content-Type: application/json" \
 'http://localhost:8080/graphql?query=?query=query($where:Filter){testCollection(where:$where){items{title}}}&variables={"draft":true}

**Comment by shuston** [comment]
Usually the query is sent as the body of a POST

**Reply by shuston** [reply]
If you can run GraphQL Playground or GraphiQL on your API you'll be able to see the content of the query and variables. GraphQL Playground will also help you enter a valid query. Maybe GraphiQL will too - I've not played with that.

**Comment by paulbakker** [comment, marked as answer]
The framework only supports POST requests. Note that it supports both `application/json` and `application/graphql`. The latter is really nice when using something like curl, because you don't have to quote/escape things.

```
curl -X POST --location "http://localhost:8080/graphql" \
    -H "Content-Type: application/graphql" \
    -d "{
  hello
}"
```


Besides curl, you can also use the HTTP Request feature in Intellij, and of course there is `/graphiql`.

**Reply by Ericnunez** [reply]
thank you!
